icu4c/source/test/intltest/caltest.cpp

```diff
@@ -4086,4 +4088,69 @@ void CalendarTest::TestChineseCalendarMapping() {
 
 #endif /* #if !UCONFIG_NO_FORMATTING */
 
+void CalendarTest::TestChineseCalendarMonthInSpecialYear(void) {
+    UErrorCode status = U_ZERO_ERROR;
+    GregorianCalendar gc(status);
+    ChineseCalendar cal(Locale::getRoot(), status);
+    if (failure(status, "Constructor failed")) return;
+    struct TestCase {
+      int32_t gyear;
+      int32_t gmonth;
+      int32_t gdate;
+      int32_t cmonth;
+      int32_t cdate;
+      bool cleapmonth;
+    } cases[] = {
+        // Gregorian             Chinese Calendar
+        // First some recent date
+        // From https://www.hko.gov.hk/tc/gts/time/calendar/pdf/files/2022.pdf
+        { 2022, UCAL_DECEMBER, 15, UCAL_NOVEMBER, 22, false},
+        // From https://www.hko.gov.hk/tc/gts/time/calendar/pdf/files/2023.pdf
+        { 2023, UCAL_MARCH, 21, UCAL_FEBRUARY, 30, false},
+        { 2023, UCAL_MARCH, 22, UCAL_FEBRUARY, 1, true},
+        // We know there are some problematic year, especially those involved
+        // the rare cases of M11L and M12L.
+        // Check 1889, 1890, 2033
+        // 2033 has M11L
+        // From https://www.hko.gov.hk/tc/gts/time/calendar/pdf/files/2033.pdf
+        { 2033, UCAL_DECEMBER, 21, UCAL_NOVEMBER, 30, false},
+        { 2033, UCAL_DECEMBER, 22, UCAL_NOVEMBER, 1, true},
+        // Here are the date we get from multiple external sources
+        // https://ytliu0.github.io/ChineseCalendar/index_chinese.html
+        // https://ytliu0.github.io/ChineseCalendar/table_period.html?period=qing
+        { 1890, UCAL_JANUARY, 1, UCAL_DECEMBER, 11, false},
+        { 1890, UCAL_JANUARY, 20, UCAL_DECEMBER, 30, false},
+        { 1890, UCAL_JANUARY, 21, UCAL_JANUARY, 1, false},
+        { 1890, UCAL_FEBRUARY, 1, UCAL_JANUARY, 12, false},
+        { 1890, UCAL_FEBRUARY, 19, UCAL_FEBRUARY, 1, false},
+        { 1890, UCAL_MARCH, 1, UCAL_FEBRUARY, 11, false},
+        { 1890, UCAL_MARCH, 21, UCAL_FEBRUARY, 1, true},
+        { 1890, UCAL_APRIL, 1, UCAL_FEBRUARY, 12, true},
+        { 1890, UCAL_APRIL, 18, UCAL_FEBRUARY, 29, true},
+        { 1890, UCAL_APRIL, 19, UCAL_MARCH, 1, false},
+        { 1890, UCAL_APRIL, 20, UCAL_MARCH, 2, false},
```

Comment:
So the issue with 1890 is that it maps to a Chinese year that has a leap month after the 12th month of the year, something that hasn't happened since and won't happen again for a very long time.  I'm guessing that this test checks for the _correct_ results, and that you're making it as "expected to fail" for now until the actual problem is fixed?  What are we _actually_ getting right now?

Reply:
The Lunar year which majorty part fall into 1889 and the early part of 1890 should have no leap months, but currently ICU calculate and show there is a Leap month after the 12th month and before the first month of the Chinese Calendar which overlapping most of the 1890 year in Gregorian. 

We use the value from https://ytliu0.github.io/ChineseCalendar/table_period.html?period=qing and  https://ytliu0.github.io/ChineseCalendar/index_chinese.html for the correct value. The same results were given by many several other sites not just his one. 

There should be a Leap month after the 2nd month of the Chinese Calendar year mostly overlapping with 1890 and should have no leap month in the Chinese Calendar year mostly overlapping with 1889.

Currently, if I run ./intltest  format/CalendarTest/TestChineseCalendarMonthInSpecialYear  -K
(-K to run the known issue)
I got
```
=== Handling test: format/CalendarTest/TestChineseCalendarMonthInSpecialYear: ===
   format {
      CalendarTest {
         TestChineseCalendarMonthInSpecialYear {
         Fail: Gregorian(1890/1/21) should be Chinese 1/1 but got 12L/1
         Fail: Gregorian(1890/2/1) should be Chinese 1/12 but got 12L/12
         Fail: Gregorian(1890/2/19) should be Chinese 2/1 but got 1/1
         Fail: Gregorian(1890/3/1) should be Chinese 2/11 but got 1/11
         Fail: Gregorian(1890/3/21) should be Chinese 2L/1 but got 2/1
         Fail: Gregorian(1890/4/1) should be Chinese 2L/12 but got 2/12
         Fail: Gregorian(1890/4/18) should be Chinese 2L/29 but got 2/29
      
         } ERRORS (7) in TestChineseCalendarMonthInSpecialYear
      
   
      } ERRORS (7) in CalendarTest
   

   } ERRORS (7) in format


--------------------------------------
Errors in total: 7.
            TestChineseCalendarMonthInSpecialYear
         CalendarTest
      format

```